prov/shm/src/smr_init.c

```diff
@@ -134,7 +135,11 @@ static int smr_getinfo(uint32_t version, const char *node, const char *service,
 
 static void smr_fini(void)
 {
-	/* yawn */
+	struct smr_ep_names *ep_name;
+	dlist_foreach_container(&ep_names.entry, struct smr_ep_names,
+				ep_name, entry) {
+		free(ep_name);
```

Comment:
I believe this will result in use after free.  We need dlist_foreach_container_safe.